src/components/MyEvents.js

```diff
@@ -21,9 +21,30 @@ class MyEvents extends Component {
 		}
 	}
 
+	validateForm = () => {
+		return new Promise((resolve, reject) => {
+			let phoneNo = this.refs.phone.value
+			if (phoneNo.length !== 10) {
+				alert('Please enter a 10 digit phone number')
+				reject()
+			} else if (isNaN(phoneNo)) {
+				alert('Value should be numeric.')
+				phoneNo = ""
+				reject()
+			} else {
+				resolve()
+			}
+		})
+	}
+
 	addPhone(event) {
 		event.preventDefault();
-		this.props.saveUserPhone(this.refs.phone.value);
+		this.validateForm()
+			.then(() => {
+				this.props.saveUserPhone(this.refs.phone.value);
+				this.refs.phone.value = ""
+			})
+			.catch(() => console.log('Resolve error.'))		
```

Comment:
How will the user know that he has entered a wrong number?

A better approach can be having a green, red or transparent border for success, error and "typing" respectively.